File: debug-mcp-vscode/src/debugger/CommandExecutor.ts
import * as vscode from "vscode";
import { BridgeMessage, MessageTypes } from "../bridge/MessageProtocol";

export class CommandExecutor {
  async handle(message: BridgeMessage) {
    if (message.type === MessageTypes.AgentContinue) {
      await vscode.commands.executeCommand("workbench.action.debug.continue");
    }
    if (message.type === "agent_step_over") {
      await vscode.commands.executeCommand("workbench.action.debug.stepOver");
    }
    if (message.type === "agent_step_into") {
      await vscode.commands.executeCommand("workbench.action.debug.stepInto");
    }
    if (message.type === "agent_step_out") {
      await vscode.commands.executeCommand("workbench.action.debug.stepOut");
    }
  }
}

File: debug-mcp-idea/src/main/kotlin/debugger/CommandExecutor.kt
package debugger

import bridge.BridgeMessage
import bridge.MessageTypes
import com.intellij.openapi.project.Project
import com.intellij.xdebugger.XDebuggerManager

class CommandExecutor(private val project: Project) {
    fun handle(message: BridgeMessage) {
        val session = XDebuggerManager.getInstance(project).currentSession ?: return
        when (message.type) {
            MessageTypes.AgentContinue -> session.resume()
            "agent_step_over" -> session.stepOver(false)
            "agent_step_into" -> session.stepInto()
            "agent_step_out" -> session.stepOut()
        }
    }
}
